internal/controller/modelregistry_controller.go

```diff
@@ -240,8 +240,10 @@ func (r *ModelRegistryReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&modelregistryv1alpha1.ModelRegistry{}).
 		Owns(&corev1.Service{}).
 		Owns(&corev1.ServiceAccount{}).
-		Owns(&appsv1.Deployment{})
+		Owns(&appsv1.Deployment{}).
+		Owns(&rbac.Role{})
 	if r.IsOpenShift {
+		builder = builder.Owns(&rbac.RoleBinding{})
```

Comment:
I understand this is here as it aligns to:

https://github.com/opendatahub-io/model-registry-operator/blob/c8e16cf5d020df1cf25ef1a94e506b779fdc4869/internal/controller/modelregistry_controller.go#L309-L319